homeassistant/components/camera/amcrest.py

```diff
@@ -64,26 +82,84 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
             notification_id=NOTIFICATION_ID)
         return False
 
-    add_devices([AmcrestCam(config, data)])
+    add_devices([AmcrestCam(hass, config, data)])
     return True
 
 
 class AmcrestCam(Camera):
     """An implementation of an Amcrest IP camera."""
 
-    def __init__(self, device_info, data):
+    def __init__(self, hass, device_info, data):
         """Initialize an Amcrest camera."""
         super(AmcrestCam, self).__init__()
+        self._base_url = '%s://%s:%s/cgi-bin' % (
+            'http',
+            device_info.get(CONF_HOST),
+            device_info.get(CONF_PORT)
+        )
         self._data = data
+        self._hass = hass
         self._name = device_info.get(CONF_NAME)
         self._resolution = RESOLUTION_LIST[device_info.get(CONF_RESOLUTION)]
+        self._stream_source = STREAM_SOURCE_LIST[
+            device_info.get(CONF_STREAM_SOURCE)
+        ]
+        self._token = self._auth = aiohttp.BasicAuth(
```

Comment:
Did you try to see if the self._data.camera._token attribute works or it does really requires a aiohttp?
Also, why do you need the self._auth if this is not being used on the code anywhere else?